File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        count = collections.Counter(s)

        for char in t:
            if char not in count:
                return False
            elif count[char] == 1:
                del count[char]
            else:
                count[char] -= 1
        
        return len(count.keys()) == 0 
        

File: 0242-valid-anagram/0242-valid-anagram.go
func isAnagram(s string, t string) bool {
    count := make(map[rune]int)

    for _, char := range s {
        count[char]++
    }

    for _, char := range t {
        if _, ok := count[char]; !ok {
            return false
        } else if count[char] == 1{
            delete(count, char)
        } else {
            count[char]--
        }
    }

    return len(count) == 0
}